### ISSUE TITLE
Null ref: T2.CanvasMenu.MakeCurrentMenuItemFadeOut

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#995]
Auto report, 0.3.1, feedback ID 20220111-FRFLV

```
6:40:48, Frame 123562, LOG »  InventoryItemStruct.SellToMerchant: Transfering item Master989 to merchant Master24
6:40:48, Frame 123562, LOG »  InventoryItemStruct.TransferToMerchant: Transfering item to merchant... about to make merchant acquire item


6:41:30, Frame 125181, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.CanvasMenu.MakeCurrentMenuItemFadeOut ()
T2.CanvasMenu.Update ()

GameStates: RompingThroughDungeon, PlayerViewingCreatureInventory
OnlineGameInfo: Device is offline.
No previous server nor online game info.
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 0 / 53
RoomModifiers: None
YouHaveDiedState: Null
SystemPlayer InputDevice: Keyboard
HumanPlayer 1 InputDevice: Keyboard
```

**Comment by ChrisNZL** [comment]
Fixed in 0.3.1a. Added null checks within `CanvasMenu.MakeCurrentMenuItemFadeOut`.

**Comment by ChrisNZL** [comment]
Re-opening. Still happening a lot in 0.3.1a.

**Comment by ChrisNZL** [comment]
Should be fixed now in 0.3.2. Added a null check I'd missed previously.

```csharp
void MakeCurrentMenuItemFadeOut () {
	Category _category = this._Category;
	if (_category.Exists()) {
		CategoryItem selectedItem = _category.SelectedItem;
		if (selectedItem.Exists()) {
			CanvasMenuItem menuItem = selectedItem.canvasMenuItem;
			if (menuItem.Exists()) {
				CanvasMenuItemSelectedBorder selectedBorder = menuItem.selectedBorder;
				if (selectedBorder.Exists()) {
					selectedBorder.FadeOut();
				}

				Scaler textScaler = menuItem.textScaler;
				if (textScaler.Exists()) {
					textScaler.ResetScale();
				}
			}
		}
	}
}
```

Also added logging for `CanvasMenu.CurrentCategory`.
